web-module/src/main/webapp/resources/structEditJS/dateTime/celements-date-time-field.js

```diff
@@ -0,0 +1,461 @@
+      },
+
+      getTimeValue : function() {
+        var _me = this;
+        var dateTimeValues = $F(_me._hiddenDateTimeField).split(' ');
+        var timeValue = dateTimeValues[1] || "00:00";
+        console.log('getTimeValue: ', timeValue);
+        return timeValue;
+      },
+
+      getDateValue : function() {
+        var _me = this;
+        var dateTimeValues = $F(_me._hiddenDateTimeField).split(' ');
+        var dateValue = dateTimeValues[0];
+        console.log('getDateValue: ', dateValue);
+        return dateValue;
+      },
+
+      _updateVisibleFromHidden : function() {
+        var _me = this;
+        var dateValue = _me.getDateValue();
+        _me._inputDateField.setValue(dateValue);
+        var timeValue = _me.getTimeValue();
+        _me._inputTimeField.setValue(timeValue);
+        console.log("_updateVisibleFromHidden", _me._hiddenDateTimeField, dateValue, timeValue);
+        _me._updateHiddenFromVisible();
+      },
+
+      _updateHiddenFromVisible : function() {
+        var _me = this;
+        var dateValue = _me._inputDateField.getValue();
+        var timeValue = _me._inputTimeField.getValue();
+        // if (_me._isAllDayFnc()) {
+          // timeValue = "00:00";
+        // }
+        var dateTimeValues = dateValue + " " + timeValue;
+        _me._hiddenDateTimeField.value = dateTimeValues;
+        console.log("_updateHiddenFromVisible", dateTimeValues);
+      },
+
+      _allDayChangedHandler : function(event) {
+        var _me = this;
+        _me._updateHiddenFromVisible();
+      }
+
+    });
+  }
+
+  if (typeof CelementsDateTimeField === 'undefined') {
+    class CelementsDateTimeField extends HTMLElement {
+      constructor() {
+        super();
+        const _me = this;
+        _me.attachShadow({mode: 'open'});
+        _me.addCssFilesToParent();
+        _me.addCssFiles();
+        _me.addInputFields();
+        _me.addPickerIcons();
+      }
+
+      addCssFilesToParent() {
+        //HACK be sure to load the glyphicons-halflings.css in the html-page too.
+        //HACK Because font-face will not work in shadow dom otherwise.
+        const cssFiles = [ 'celRes/images/glyphicons-halflings/css/glyphicons-halflings.css' ];
+        cssFiles.forEach(function(cssFile) {
+          const cssElem = new Element('link', {
+            'rel' : 'stylesheet',
+            'media' : 'all',
+            'type' : 'text/css',
+            'href' : '/file/resources/' + cssFile + '?version=' + versionTimeStamp
+          });
+          document.head.append(cssElem);
+        });
+      }
+
+      addCssFiles() {
+        const _me = this;
+        //HACK be sure to load the glyphicons-halflings.css in the html-page too.
+        //HACK Because font-face will not work in shadow dom otherwise.
+        const cssFiles = [ 'celRes/images/glyphicons-halflings/css/glyphicons-halflings.css',
+          'celJS/jquery%2Ddatetimepicker/jquery.datetimepicker.css'

```

Comment:
can't these be links in `DateTimeView.vm`?

Reply:
Must be added to the shadowDom. CSS from main document have no influence on the shadowDOM.

Reply:
ok, but can the links still be put into html and collected here by selecting all `link[rel="stylesheet"]` within the innerHtml?

I think it could soften the hack a little since changes/additions to the file list could be made in html as usual.

Reply:
I do not see the benefit of this. There are only a few CSS files needed and it is very unlikely to change without the js-file needs changes too.

Reply:
More over it is an implementation detail of the JS-WebComponent and must not be put to the client

Reply:
ok, I understand. Thanks for entertaining the idea.